### ISSUE TITLE
Print graphql queries to logs

### ISSUE THREAD

**Issue by shykes** [dagger/dagger#3561]
## Problem

When writing pipelines using the [Go SDK](https://docs.dagger.io/sdk/go), it's not always clear exactly what the resulting graphql query (or queries) will look like. This is especially needed when debugging, and learning Dagger.

## Solution

Change the engine logger so that it prints raw graphql queries, rather than only buildkit-specific information.

**Comment by shykes** [comment]
Would this qualify as size/small @vito @sipsma ?

**Comment by sipsma** [comment]
> Would this qualify as size/small

We'd have to hook into the buildkit progress logs. There's a chance that's easy for someone familiar with those, but even then it may be easier said than done. If someone's not familiar with buildkit progress logs, then no, not small probably.

Or other option is to abstract on top of buildkit progress logs and use our own logging system (similar to Europa). Good thing to do, not small.

tl;dr no, I don't think so. Happy to be wrong though

**Comment by shykes** [comment]
What if you had to choose between two completely different log streams: 1) buildkit or 2) graphql, and we made no attempt to integrate the two. You just get one bytestream or the other. Would that make it simpler?

For example in my case I would gladly turn off buildkit logs in exchange for a clean log of graphql queries as they are being routed - more relevant to my debugging.

**Comment by sipsma** [comment]
> What if you had to choose between two completely different log streams: 1) buildkit or 2) graphql, and we made no attempt to integrate the two. You just get one bytestream or the other. Would that make it simpler?

I think so. It would solve the problem of the logs otherwise being interleaved (on char-by-char basis). You could add an optional to `dagger.Connect`, which would then enable printing from codegen client.

So yeah pretty small then.

Only con left I can think of is backcompat in future if we change logging more, but I am inclined to not care, too minor relative to benefit.

**Comment by shykes** [comment]
This is addressed by new TUI. Yay!
